core/invoker/src/main/scala/whisk/core/invoker/Invoker.scala

```diff
@@ -188,5 +188,20 @@ object Invoker {
     BasicHttpService.startHttpService(new BasicRasService {}.route, port)(
       actorSystem,
       ActorMaterializer.create(actorSystem))
+
+    Signal.handle(
+      new Signal("USR2"),
+      new SignalHandler() {
+        override def handle(signal: Signal) = {
+          signal.getName match {
+            case "USR2" =>
+              logger.info(this, s"Stopping health scheduler")
+              actorSystem.stop(healthScheduler)
+              while (true) {}
```

Comment:
For a graceful shutdown, we would need to make sure user' containers are not running and possibly check that the queue is empty.

Reply:
By queue do you mean the internal message feed or the kafka topic? The topic doesn't need to be empty as that can take an indeterminate amount of time --- bringing up another invoker to take over the topic would address the messages held in the topic.

Reply:
For the Kafka topic, I mean. Need to find a way to make the invoker stop consuming messages from the topic. Will take a look.

Reply:
For the above, might just be able to shutdown the `activationFeed` actor.

Reply:
I thought the feed had a shutdown hook. It is worth adding one.

Reply:
Looks like there is a shutdown hook for `containerFactory`. Perhaps maintenance mode is not necessary if we have a graceful shutdown.